### PR TITLE
speedup test_listen_and_serv_op

### LINKED ISSUE #11127: test_listen_and_serv_op timeout
CI job link: https://paddleci.ngrok.io/viewLog.html?buildId=37582&buildTypeId=Paddle_PrCi&tab=buildLog

### PR DESCRIPTION
**Pull request by Yancey1989**
Fixed #11127

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_listen_and_serv_op.py [hunk 1]
```diff
@@ -62,47 +55,47 @@ def run_pserver(use_cuda, sync_mode, ip, port, trainer_count, trainer_id):
...
+        os.kill(pid, signal.SIGINT)
 
         # run pserver on CPU in async mode
-        self._start_pserver(False, False)
+        pid = self._start_pserver(False, False)
+        self._wait_ps_ready(pid)
 
         # raise SIGTERM to pserver
-        self._raise_signal(os.getpid(), signal.SIGTERM)
+        os.kill(pid, signal.SIGINT)
```
Comment (velconia) [inline python/paddle/fluid/tests/unittests/test_listen_and_serv_op.py]: Signal shoule be ```SIGTERM``` here

File: python/paddle/fluid/tests/unittests/test_listen_and_serv_op.py [hunk 2]
```diff
@@ -62,47 +55,47 @@ def run_pserver(use_cuda, sync_mode, ip, port, trainer_count, trainer_id):
...
+            args=(use_cuda, sync_mode, self.ip, self.port, self.trainers,
                   self.trainer_id))
         p.start()
+        return p.pid
+
+    def _wait_ps_ready(self, pid):
+        retry_times = self.ps_timeout
+        while True:
+            time.sleep(1)
+            assert retry_times >= 0, "wait ps ready failed"
```
Comment (velconia) [inline python/paddle/fluid/tests/unittests/test_listen_and_serv_op.py]: The ```assert``` line and the ```time.sleep(1)``` line can be exchanged to detect the timeout ealier

File: python/paddle/fluid/tests/unittests/test_listen_and_serv_op.py [hunk 3]
```diff
@@ -62,47 +55,47 @@ def run_pserver(use_cuda, sync_mode, ip, port, trainer_count, trainer_id):
...
-            args=(use_cuda, sync_mode, self.ip, self.port, self.trainer_count,
+            args=(use_cuda, sync_mode, self.ip, self.port, self.trainers,
                   self.trainer_id))
         p.start()
+        return p.pid
+
+    def _wait_ps_ready(self, pid):
+        retry_times = self.ps_timeout
+        while True:
+            time.sleep(1)
```
[the remark this answers is not in the document]
Comment (Yancey1989) [reply python/paddle/fluid/tests/unittests/test_listen_and_serv_op.py]: Sure, but the pserver would cost about 1 seconds until become ready. BTW, the most time-consuming place is to load and initialize the cuda libraries.

File: python/paddle/fluid/tests/unittests/CMakeLists.txt [hunk 4]
```diff
@@ -48,5 +48,7 @@ foreach(TEST_OP ${TEST_OPS})
 endforeach(TEST_OP)
 py_test_modules(test_warpctc_op MODULES test_warpctc_op ENVS FLAGS_warpctc_dir=${WARPCTC_LIB_DIR} SERIAL)
 py_test_modules(test_dist_train MODULES test_dist_train SERIAL)
-# tests that need to be done in fixed timeout
-set_tests_properties(test_listen_and_serv_op PROPERTIES TIMEOUT 20)
+# FIXME(Yancey1989): this test would cost much more time on CUDAPlace
+# since load cudnn libraries, so we use a longer timeout to make this
+# unit test stability.
+set_tests_properties(test_listen_and_serv_op PROPERTIES TIMEOUT 30)
```
[the remark this answers is not in the document]
Comment (Yancey1989) [reply python/paddle/fluid/tests/unittests/CMakeLists.txt]: Yes, in this unit test, pserver would be killed when pserver is ready.

File: python/paddle/fluid/tests/unittests/test_listen_and_serv_op.py (hunk 1 above)
Comment (Yancey1989) [reply python/paddle/fluid/tests/unittests/test_listen_and_serv_op.py]: Done.

File: python/paddle/fluid/tests/unittests/test_listen_and_serv_op.py (hunk 2 above)
Comment (Yancey1989) [reply python/paddle/fluid/tests/unittests/test_listen_and_serv_op.py]: Done.

**Pull request merged**